pydantic/dataclasses.py

```diff
@@ -103,76 +105,74 @@ def __validate__(cls: Type['DataclassT'], v: Any) -> 'DataclassT':
 
 _T = TypeVar('_T')
 
-if sys.version_info >= (3, 10):
-
-    @dataclass_transform(kw_only_default=True, field_specifiers=(Field, FieldInfo))
-    @overload
-    def dataclass(
-        *,
-        init: bool = True,
-        repr: bool = True,
-        eq: bool = True,
-        order: bool = False,
-        unsafe_hash: bool = False,
-        frozen: bool = False,
-        config: Union[ConfigDict, Type[object], None] = None,
-        validate_on_init: Optional[bool] = None,
-        kw_only: bool = ...,
-    ) -> Callable[[Type[_T]], 'DataclassClassOrWrapper']:
-        ...
-
-    @dataclass_transform(kw_only_default=True, field_specifiers=(Field, FieldInfo))
-    @overload
-    def dataclass(
-        _cls: Type[_T],
-        *,
-        init: bool = True,
-        repr: bool = True,
-        eq: bool = True,
-        order: bool = False,
-        unsafe_hash: bool = False,
-        frozen: bool = False,
-        config: Union[ConfigDict, Type[object], None] = None,
-        validate_on_init: Optional[bool] = None,
-        kw_only: bool = ...,
-    ) -> 'DataclassClassOrWrapper':
-        ...
-
-else:
-
-    @dataclass_transform(kw_only_default=True, field_specifiers=(Field, FieldInfo))
-    @overload
-    def dataclass(
-        *,
-        init: bool = True,
-        repr: bool = True,
-        eq: bool = True,
-        order: bool = False,
-        unsafe_hash: bool = False,
-        frozen: bool = False,
-        config: Union[ConfigDict, Type[object], None] = None,
-        validate_on_init: Optional[bool] = None,
-    ) -> Callable[[Type[_T]], 'DataclassClassOrWrapper']:
-        ...
-
-    @dataclass_transform(kw_only_default=True, field_specifiers=(Field, FieldInfo))
-    @overload
-    def dataclass(
-        _cls: Type[_T],
-        *,
-        init: bool = True,
-        repr: bool = True,
-        eq: bool = True,
-        order: bool = False,
-        unsafe_hash: bool = False,
-        frozen: bool = False,
-        config: Union[ConfigDict, Type[object], None] = None,
-        validate_on_init: Optional[bool] = None,
-    ) -> 'DataclassClassOrWrapper':
-        ...
-
-
-@dataclass_transform(kw_only_default=True, field_specifiers=(Field, FieldInfo))
+if TYPE_CHECKING:
+    if sys.version_info >= (3, 10):
```

Comment:
sorry if I'm being dumb, why do we need different logic for 3.10 and before?

Reply:
`3.10` adds an additional argument `kw_only`.